### ISSUE TITLE
Any chance of putting this wonderful app on F-Droid store?

### ISSUE THREAD

**Issue by defphil** [fibelatti/pinboard-kotlin#11]
I would really appreciate it (and I am sure many others).

**Comment by fibelatti** [comment]
Hi @defphil, thanks for the suggestion. I will look into doing it.

**Comment by grrrrr** [comment]
I'll second this request, it would be very useful to have on f-droid given that the other android app, pindroid has stalled in development

**Comment by fibelatti** [comment]
> Hi @fibelatti
> 
> If f-droid is not an options, would it be possible to release APK on gitbub? It might allow for inclusion with @IzzySoft's repo which can be used with f-droid https://apt.izzysoft.de/fdroid/

Hey @grrrrr, thanks for suggesting this alternative, I know it's been _a good while_ since adding the app to F-Droid was first suggested. I've looked into it a while back but never managed to go through the full setup required, so this alternative might facilitate things.

I still need to make a few changes to the app because I'm use some play services (to auto-update and prompt for app store reviews). I'll either have to remove those or build a different app flavour that's specific to F-Droid. I'll prioritise this and see what it takes to add it to the repo

**Comment by fibelatti** [comment]
@IzzySoft I have just released [v2](https://github.com/fibelatti/pinboard-kotlin/releases/tag/v2.0) which should be suitable for your repository. I have attached the signed `.apk` to the release on GitHub, is there anything else I should be doing to have it listed?

**Comment by IzzySoft** [comment]
> I have just released [v2](https://github.com/fibelatti/pinboard-kotlin/releases/tag/v2.0) which should be suitable for your repository.

Confirmed: it is :smiley:

> is there anything else I should be doing to have it listed?

Just to wait until it [shows up here](https://apt.izzysoft.de/packages/com.fibelatti.pinboard/) with the next sync around 6 pm UTC :smiley: If you want to define how your app is presented, you could establish Fastlane structures here in your app's repo (be welcome to use my [Fastlane Cheat Sheet](https://gitlab.com/-/snippets/1895688) for guidance with that). And of course you're also welcome to [pick a badge](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) to link to your app's new presence :wink:

Thanks a lot!

**Comment by fibelatti** [comment]
@IzzySoft awesome, thank you! I'll look into setting up fastlane and adding a badge to the repository later today 🚀

**Comment by IzzySoft** [comment]
Fantastic! Please givve me a ping here once Fastlane is ready – so I cross-check, enable it on my end and (if you link me that RFP I saw a few hours again) also kick the issuebot for another scan at F-Droid.org :wink:

**Comment by fibelatti** [comment]
@IzzySoft I've setup the metadata for fastlane now and published a [new release](https://github.com/fibelatti/pinboard-kotlin/releases/tag/v2.0-fdroid) just in case since F-Droid doesn't build from HEAD. This new tag should also fix the Gradle wrapper issue from the [RFP](https://gitlab.com/fdroid/rfp/-/issues/2527) 👀 

I'm gonna look into the badges next, thanks for your support

**Comment by fibelatti** [comment]
@IzzySoft no problem! As always, thanks a mil for your support!

**Comment by IzzySoft** [comment]
> couldn't think of a better way to reach out!

Totally fine this way :smiley: 

> could you please enable that on your side

That was already done on 2023-10-02. Changes on your end would be pulled along with new releases.

> I'm about to include `com.google.android.gms:play-services-oss-licenses`

Bad idea. The "oss" in that name is misleading: it only processes FOSS-licenses but is not fully FOSS itself (but will drag in proprietary dependencies like GMS) – those AS-terms are proprietary, too (limiting freedoms). So yes, I'd say that is a problem. Mind to pick [one of the alternatives](https://gitlab.com/-/snippets/1896503#license-dialogs) I've collected for you? :wink:

**Comment by fibelatti** [comment]
@IzzySoft you're a gentleman sir! Thanks for enabling the metadata change and for linking the alternatives! I'll look into them before submitting the next update.

I've also released a new app this weekend but it already contains the play library so I'll remove it first before submitting it to IzzyOnDroid as well.

Thanks a mil Izzy!

**Comment by IzzySoft** [comment]
You are very welcome! Just give me a ping when your new app is ready. Don't be surprised if I then respond that it's already listed :see_no_evil: 

Some background:

* there's a quality check script running here once a month, a.o. checking if some repo has Fastlane added while it's not available here, or whether it is available on both ends but not fully enabled here. That's how your fastlane structures were already added in October: my tool reported it to me.
* I have some other tool watching out for new apps on Codeberg, Gtihub, GitLab which are neither listed at F-Droid nor in my repo. That's how many apps come to my repo.
* there are several more things automated here. Like, had you added that library, as soon as the new APK was pulled here an alert had popped up notifying me. That's e.g. how one app got removed from my repo today, for exceeding those thresholds by far (7 offending libraries is simply too much)

Not all is automated, though. So don't you worry I'd get bored – I won't :see_no_evil:
